cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -21,19 +21,21 @@ type SimulateTransactionCost struct {
 	MemoryBytes     uint64 `json:"memBytes,string"`
 }
 
-type SimulateTransactionResult struct {
-	Auth      []string `json:"auth"`
-	Events    []string `json:"events"`
-	Footprint string   `json:"footprint"`
-	// TODO: update documentation and review field name
-	XDRs []string `json:"xdrs"`
+// SimulateHostFunctionResult contains the simulation result of each HostFunction within each InvokeHostFunctionOp
+type SimulateHostFunctionResult struct {
+	Auth []string `json:"auth"`
+	XDR  string   `json:"xdr"`
 }
 
 type SimulateTransactionResponse struct {
-	Error        string                      `json:"error,omitempty"`
-	Results      []SimulateTransactionResult `json:"results,omitempty"`
-	Cost         SimulateTransactionCost     `json:"cost"`
-	LatestLedger int64                       `json:"latestLedger,string"`
+	Error string `json:"error,omitempty"`
+	// TODO: update documentation and review field names
+	TransactionData string                       `json:"transaction_data"` // SorobanTransactionData XDR in base64
+	Events          []string                     `json:"events"`           // DiagnosticEvent XDR in base64
+	MinResourceFee  int64                        `json:"minResourceFee,string"`
+	Results         []SimulateHostFunctionResult `json:"results,omitempty"`
+	Cost            SimulateTransactionCost      `json:"cost"`
```

Comment:
is `Cost` providing anything unique, it appeared like `TransactionData Resources` has the same attributes with further detail?

Reply:
That is true ... maybe we can remove them. @tsachiherman thoughts?

Reply:
I'll look into that tomorrow morning, but I think that we shouldnt have any duplicate info in here.

Reply:
yes, absolutely. I don't see much value keeping this if that information can be obtained via the SorobanTransactionData.

Reply:
Alright, I will remove them

Reply:
Actually, the soroban transaction data doesn't contain memory consumption and the instructions it contains are the resources it allows in for execution (which contain a leeway), so I think I will leave it.

Reply:
ok, will retain reference to `cost` on the soroban-client simulation response model.